### PR TITLE
Illustration of the observable part of the QUIC handshake

### LINKED ISSUE #35: Handshake Illustration
The manageability document should illustrate all the various packet sequences in a handshake, preferably with pretty pictures.

### PR DESCRIPTION
**Pull request by britram**
fix #35

### CODE REVIEW

File: draft-ietf-quic-manageability.md [hunk 1]
```diff
@@ -181,17 +181,192 @@ used to separate QUIC packets. The length header field is variable length and
...
+A typical handshake starts with the client sending of a QUIC Client Hello
+datagram as shown in {{fig-client-hello}}, which elicits a QUIC Server Hello
+datagram as shown in {{fig-server-hello}} typically containing three packets:
+an Initial packet with the Server Hello, a Handshake packet with the rest of
+the server's side of the TLS handshake, and initial 1-RTT data, if present.
+
+The content of QUIC Initial packets are encrypted using Initial Secrets, which
+are derived from a per-version constant and the client's destination connection
+ID; they are therefore observable by any on-path device that knows the
+per-version constant; we therefore these as visible in our illustration. The
```
[the remark this answers is not in the document]
Comment (britram) [reply draft-ietf-quic-manageability.md]:
```suggestion
per-version constant; we therefore consider these as visible in our illustration. The
```

File: draft-ietf-quic-manageability.md [hunk 2]
```diff
@@ -181,17 +181,192 @@ used to separate QUIC packets. The length header field is variable length and
...
+
+To illustrate the information visible in the QUIC wire image during the
+handshake, we first show the general communication pattern visible in the UDP
+datagams containing the QUIC handshake, then examine each of the datagrams in
+detail.
+
+In the nominal case, the QUIC handshake can be recognized on the wire through at
+least four datagrams we'll call "QUIC Client Hello", "QUIC Server Hello", and
+"Initial Completion", and "Handshake Completion", for purposes of this
+illustration, sent as shown in {{fig-handshake}}. As shown here, the client can
```
[the remark this answers is not in the document]
Comment (britram) [reply draft-ietf-quic-manageability.md]:
```suggestion
illustration, as shown in {{fig-handshake}}. As shown here, the client can
```

File: draft-ietf-quic-manageability.md [hunk 3]
```diff
@@ -181,17 +181,201 @@ used to separate QUIC packets. The length header field is variable length and
...
+Client                                    Server
+  |                                          |
+  +----QUIC Client Hello-------------------->|
+  +----(zero or more 0RTT)------------------>|
+  |                                          |
+  |<--------------------QUIC Server Hello----+
+  |<---------(1RTT encrypted data starts)----+
+  |                                          |
+  +----Initial Completion------------------->|
+  +----(optional 0RTT completion)----------->|
```
[the remark this answers is not in the document]
Comment (mirjak) [reply draft-ietf-quic-manageability.md]: No, should probably just be "optional 0RTT continuation".

[the remark this answers is not in the document]
Comment (britram) [reply draft-ietf-quic-manageability.md]: Ah, yep. (It's still in section 6.5 of -15, which I used as a reference here). Will fix in a new PR.

File: draft-ietf-quic-manageability.md [hunk 4]
```diff
@@ -181,17 +181,201 @@ used to separate QUIC packets. The length header field is variable length and
...
++----------------------------------------------------------+<-+
+| QUIC long header (type = 0RTT, Version, DCID, SCID)    (Length)
++----------------------------------------------------------+  |
+| 0-rtt encrypted payload                                  |  |
++----------------------------------------------------------+<-+
+~~~~~
+{: #fig-client-hello-0rtt
+   title="Typical 0-RTT QUIC Client Hello datagram pattern"}
+
+In a 0-RTT QUIC Client Hello datagram, the PADDING frame is only present if
```
[the remark this answers is not in the document]
Comment (britram) [reply draft-ietf-quic-manageability.md]: hm, point. so there's a balance between completeness and usefulness of the illustration here (which is why I use "Typical" a lot). Will do a new PR to make it clear that the padding isn't always in the Initial domain.

File: draft-ietf-quic-manageability.md [hunk 5]
```diff
@@ -181,17 +181,201 @@ used to separate QUIC packets. The length header field is variable length and
...
 source and destionation connection ID. See Section 4.6 of {{QUIC-TRANSPORT}}.
 
+## The QUIC handshake {#handshake}
+
+New QUIC connections are established using a handshake, which is distinguishable
+on the wire and contains some information that can be passively observed.
+
+To illustrate the information visible in the QUIC wire image during the
+handshake, we first show the general communication pattern visible in the UDP
+datagams containing the QUIC handshake, then examine each of the datagrams in
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md]: Datagams

File: draft-ietf-quic-manageability.md [hunk 6]
```diff
@@ -181,17 +181,201 @@ used to separate QUIC packets. The length header field is variable length and
...
+
+As shown here, the client can send 0-RTT data as soon as it has sent its Client
+Hello, and the server can send 1-RTT data as soon as it has sent its Server
+Hello.
+
+~~~~~
+Client                                    Server
+  |                                          |
+  +----QUIC Client Hello-------------------->|
+  +----(zero or more 0RTT)------------------>|
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md]: Be consistent in your use of hyphens on these. QWIK uses hyphens.

File: draft-ietf-quic-manageability.md [hunk 7]
```diff
@@ -181,17 +181,201 @@ used to separate QUIC packets. The length header field is variable length and
...
+| UDP header (source and destination UDP ports)            |
++----------------------------------------------------------+
+| QUIC long header (type = Initial, Version, DCID, SCID) (Length)
++----------------------------------------------------------+  |
+| QUIC CRYPTO frame header                                 |  |
++----------------------------------------------------------+  |
+| TLS Client Hello (incl. TLS SNI)                         |  |
++----------------------------------------------------------+  |
+| QUIC PADDING frame                                       |  |
++----------------------------------------------------------+<-+
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md]: This sets a dangerous pattern.  I am going to strongly recommend that people a. Put PADDING frames before CRYPTO, and b. Split CRYPTO frames.

Also there are multiple PADDING frames.

File: draft-ietf-quic-manageability.md [hunk 8]
```diff
@@ -181,17 +181,201 @@ used to separate QUIC packets. The length header field is variable length and
...
+~~~~~
+{: #fig-client-hello title="Typical 1-RTT QUIC Client Hello datagram pattern"}
+
+The Client Hello datagram exposes version number, source and destination
+connection IDs, and information in the TLS Client Hello message, including any
+TLS Server Name Indication (SNI) present, in the clear. The QUIC PADDING frame
+is present to ensure the Client Hello datagram has a minumum size of 1200
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md]: Minimum

File: draft-ietf-quic-manageability.md [hunk 9]
```diff
@@ -181,17 +181,201 @@ used to separate QUIC packets. The length header field is variable length and
...
++------------------------------------------------------------+
+| UDP header (source and destination UDP ports)              |
++------------------------------------------------------------+
+| QUIC long header (type = Initial, Version, DCID, SCID)   (Length)
++------------------------------------------------------------+  |
+| QUIC CRYPTO frame header                                   |  |
++------------------------------------------------------------+  |
+| TLS Server Hello                                           |  |
++------------------------------------------------------------+  |
+| QUIC ACK frame (acknowledging client hello)                |  |
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md]: the ordering here is similarly problematic

File: draft-ietf-quic-manageability.md [hunk 10]
```diff
@@ -181,17 +181,201 @@ used to separate QUIC packets. The length header field is variable length and
...
-in QUIC packet headers, since alteration of header information will lead to a
-failed integrity check at the receiver, and can even lead to connection
-termination.
+As soon as the cryptographic context is established, all information in the QUIC
+header, including information exposed in the packet header, is integrity
+protected. Further, information that was sent and exposed in handshake packets
+sent before the cryptographic context was established are validated later during
+the cryptographic handshake.  Therefore, devices on path MUST NOT change any
+information or bits in QUIC packet headers, since alteration of header
+information will lead to a failed integrity check at the receiver, and can even
```
Comment (martinthomson) [inline draft-ietf-quic-manageability.md]: s/even//

**Pull request merged**